pkg/csi/service/common/commonco/k8sorchestrator/wcp_node.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sorchestrator
+
+import (
+	"context"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/vsphere-csi-driver/v3/pkg/csi/service/common"
+	"sigs.k8s.io/vsphere-csi-driver/v3/pkg/csi/service/logger"
+)
+
+var clusterComputeResourceMoIds []string
+
+// InitializeCSINodes listens on node Add events to create CSINode instance for each K8s node object.
+func (c *K8sOrchestrator) InitializeCSINodes(ctx context.Context) error {
+	log := logger.GetLogger(ctx)
+	var err error
+	clusterComputeResourceMoIds, err = common.GetClusterComputeResourceMoIds(ctx)
+	if err != nil {
+		return logger.LogNewErrorf(log, "failed to get clusterComputeResourceMoIds. Error: %v", err)
+	}
+	c.informerManager.AddNodeListener(
+		func(obj interface{}) { // Add.
+			c.createCSINode(obj)
+		},
+		nil,
+		nil)
+	log.Infof("Starting informer to listen on nodes to create CSINodes")
+	return nil
+}
+
+// createCSINode creates CSINode instances for each K8s node object.
+func (c *K8sOrchestrator) createCSINode(obj interface{}) {
+	ctx, log := logger.GetNewContextWithLogger()
+	node, ok := obj.(*corev1.Node)
+	if node == nil || !ok {
+		log.Warnf("createCSINode: unrecognized object %+v found", obj)
+		return
+	}
+
+	// Check if CreateCSINodeAnnotation is present on the node. Spherelet adds this annotation on
+	// the node to communicate to CSI driver to create CSINode instance for each node.
+	csiNodeAnnotation, exists := node.ObjectMeta.Annotations[common.CreateCSINodeAnnotation]
```

Comment:
Do spherelet add this annotation when node is added or later point of time it is updating node object and adding this annotation?